mogp_gpu/src/bindings.cu

```diff
@@ -356,23 +412,196 @@ likelihood of the hyperparameters, from the current state of the emulator.)
+	     "get the nugget_size")
+        .def("set_nugget_size", &GPParams::set_nugget_size,
+	     "set nugget_size")
+        .def("get_corr", &GPParams::get_corr,
+	     "get the transformed correlation parameters")
+        .def("set_corr", &GPParams::set_corr,
+	     "set correlation parameters using transformed values")
+        .def("get_n_corr", &GPParams::get_n_corr,
+	     "get the number of correlation parameters")
+        .def("get_cov", &GPParams::get_cov,
+	     "get the transformed covariance parameter")
+        .def("set_cov", &GPParams::set_cov,
+	     "set covariance parameter using transformed values")
+        .def("get_corr_raw", &GPParams::get_corr_raw,
+	     "get the raw correlation parameters")
+        .def("unset_data", &GPParams::unset_data,
+	     "unset the flag to say params have been set")
+        .def("data_has_been_set", &GPParams::data_has_been_set,
+	     "get the flag to say whether or not params have been set")
+        .def("test_same_shape", py::overload_cast<GPParams&>(&GPParams::test_same_shape, py::const_),
+	     "test whether two GPParams objects are the same shape")
+        .def("test_same_shape", py::overload_cast<vec&>(&GPParams::test_same_shape, py::const_),
+	     "test whether two vectors of params are the same shape");
+     ////////////////////////////////////////
+     py::class_<WeakPrior>(m, "WeakPrior")
+      .def(py::init<>())
+          .def("sample", py::overload_cast<>(&WeakPrior::sample),
+          "sample from the prior")
+          .def("sample", py::overload_cast<const BaseTransform&>(&WeakPrior::sample),
+          "sample from the prior");
+     ////////////////////////////////////////
+     py::class_<InvGammaPrior, WeakPrior>(m, "InvGammaPrior")
+      .def(py::init<double, double>())
+        .def("logp", &InvGammaPrior::logp,
+             "Calculate the log probability")
+        .def("dlogpdx", &InvGammaPrior::dlogpdx,
+             "Calculate the derivative of the log probability")
+        .def("d2logpdx2", &InvGammaPrior::d2logpdx2,
+	     "Second Derivative of log prob wrt input")
+        .def("sample_x", &InvGammaPrior::sample_x,
+	     "Sample from the distribution");
+     //////////////////////////////////////
+     py::class_<GammaPrior, WeakPrior>(m, "GammaPrior")
+          .def(py::init<double, double>())
+            .def("logp", &GammaPrior::logp,
+                 "Calculate the log probability")
+            .def("dlogpdx", &GammaPrior::dlogpdx,
+                 "Calculate the derivative of the log probability")
+            .def("d2logpdx2", &GammaPrior::d2logpdx2,
+              "Second Derivative of log prob wrt input")
+            .def("sample_x", &GammaPrior::sample_x,
+              "Sample from the distribution");
+     //////////////////////////////////////
+     py::class_<BaseTransform>(m, "BaseTransform");
+     ////////////////////////////////////////
+     py::class_<CovTransform, BaseTransform>(m, "CovTransform")
```

Comment:
I did try using static methods, but did indeed run into trouble with pybind..   There might be a way around it, but I didn't find it in the day or so that I tried.